src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -46,9 +46,26 @@ public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> call
     {
         ArgumentNullException.ThrowIfNull(callback);
 
-        _registeredCallbacks.Add(callback);
+        var serializationMode = _serializationModeHandler.GetCallbackTargetSerializationMode(callback.Target);
 
-        return new PersistingComponentStateSubscription(_registeredCallbacks, callback);
+        return RegisterOnPersisting(callback, serializationMode);
+    }
+
+    /// <summary>
+    /// Register a callback to persist the component state when the application is about to be paused.
+    /// Registered callbacks can use this opportunity to persist their state so that it can be retrieved when the application resumes.
+    /// </summary>
+    /// <param name="callback">The callback to invoke when the application is being paused.</param>
+    /// <param name="serializationMode">The <see cref="PersistedStateSerializationMode"/> to register the callback.</param>
+    /// <returns>A subscription that can be used to unregister the callback when disposed.</returns>
+    public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> callback, PersistedStateSerializationMode serializationMode)
```

Comment:
This should probably be `IComponentRenderMode` instead of `PersistedStateSerializationMode` since that's what we are telling users to use.